src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +200,64 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI) => {
+        switch (executeData.method.toLowerCase()) {
+            case HTTP_POST:
+                this.setState({ executePostData: executeData }, () => {
+                    const form: ?HTMLFormElement = document.querySelector(
+                        `#bcow-execute-form-${this.id}`,
+                    );
+
+                    if (!form) {
+                        this.executeIntegrationErrorHandler(
+                            Error('execution form unavailable'),
+                        );
+                        return;
+                    }
+
+                    form.submit();
+                });
+
+                break;
+            case HTTP_GET:
+                window.open(executeData.url, '_blank');
+                break;
+            default:
+                this.executeIntegrationErrorHandler(
+                    Error('unknown invocation type'),
```

Comment:
Message as a constant

Reply:
will this case ever get executed?

Reply:
@priyajeet if the API adds a new integration type without a client change this could happen.
@ConradJChan good point, my plan is to add an OnError() callback that the developers can pass in. They'll see the message in that case. What else do you think would be useful besides a string message?

Reply:
I'll include the error, so they can see the stack trace if they are developing.

Reply:
Not sure :) Stack trace would be helpful for developing, but even a more descriptive string like, "<method> is an unsupported open with execute integration type"

Reply:
sounds good!